cpu/nrf5x_common/periph/gpio.c

```diff
@@ -125,44 +145,62 @@ void gpio_write(gpio_t pin, int value)
 int gpio_init_int(gpio_t pin, gpio_mode_t mode, gpio_flank_t flank,
                   gpio_cb_t cb, void *arg)
 {
+    assert(_gpiote_chan_index < GPIOTE_CHAN_NUMOF);
+
     /* disable external interrupt in case one is active */
     NRF_GPIOTE->INTENSET &= ~(GPIOTE_INTENSET_IN0_Msk);
     /* save callback */
-    exti_chan.cb = cb;
-    exti_chan.arg = arg;
+    exti_chan[_gpiote_chan_index].cb = cb;
+    exti_chan[_gpiote_chan_index].arg = arg;
     /* configure pin as input */
     gpio_init(pin, mode);
     /* set interrupt priority and enable global GPIOTE interrupt */
     NVIC_EnableIRQ(GPIOTE_IRQn);
     /* configure the GPIOTE channel: set even mode, pin and active flank */
-    NRF_GPIOTE->CONFIG[0] = (GPIOTE_CONFIG_MODE_Event |
+    NRF_GPIOTE->CONFIG[_gpiote_chan_index] = (GPIOTE_CONFIG_MODE_Event |
                              (pin_num(pin) << GPIOTE_CONFIG_PSEL_Pos) |
 #ifdef CPU_MODEL_NRF52840XXAA
                              ((pin & PORT_BIT) << 8) |
 #endif
                              (flank << GPIOTE_CONFIG_POLARITY_Pos));
     /* enable external interrupt */
-    NRF_GPIOTE->INTENSET |= GPIOTE_INTENSET_IN0_Msk;
+    NRF_GPIOTE->INTENSET |= GPIOTE_INTENSET_IN0_Msk << _gpiote_chan_index;
+
+    /* associate the current pin with channel index */
+    _exti_pins[_gpiote_chan_index] = pin;
+    /* increment channel index for next pin */
+    _gpiote_chan_index++;
     return 0;
 }
 
 void gpio_irq_enable(gpio_t pin)
 {
-    (void) pin;
-    NRF_GPIOTE->INTENSET |= GPIOTE_INTENSET_IN0_Msk;
+    for (unsigned int i = 0; i < _gpiote_chan_index; ++i) {
+        if (_exti_pins[i] == pin) {
+            NRF_GPIOTE->INTENSET |= GPIOTE_INTENSET_IN0_Msk << i;
+            break;
+        }
+    }
 }
 
 void gpio_irq_disable(gpio_t pin)
 {
-    (void) pin;
-    NRF_GPIOTE->INTENCLR |= GPIOTE_INTENSET_IN0_Msk;
+    for (unsigned int i = 0; i < _gpiote_chan_index; ++i) {
+        if (_exti_pins[i] == pin) {
+            NRF_GPIOTE->INTENCLR |= GPIOTE_INTENCLR_IN0_Msk << i;
```

Comment:
There seems to be a problem when clearing the IN[n] event. When printing the register value before and after it appears that is set to 0.